app/uk/gov/nationalarchives/omega/editorial/services/ReferenceDataService.scala

```diff
@@ -31,251 +31,18 @@ import javax.inject.{ Inject, Singleton }
-    Person("2QX", "Edward VII", Some("King of Great Britain and Ireland"), Some(1841), Some(1910)),
-    Person("561", "Fanshawe, Baron, of Richmond, ", None, None, None),
-    Person("46F", "Fawkes, Guy", None, Some(1570), Some(1606)),
-    Person("2JN", "George, David Lloyd", Some("1st Earl Lloyd George of Dwyfor"), Some(1863), Some(1945)),
-    Person("34X", "Halley, Edmund", None, Some(1656), Some(1742)),
-    Person("2TK", "Halifax, ", Some(""), None, None),
-    Person("39T", "Irvine, Linda Mary", Some("nee Newton"), Some(1928), None),
-    Person("4", "Jack the Ripper, ", None, Some(1888), Some(1888)),
-    Person("4FF", "Keay, Sir Lancelot Herman", Some("Knight"), Some(1883), Some(1974)),
-    Person("ST", "Lawson, Nigel", Some("Baron Lawson of Blaby"), Some(1932), None),
-    Person(
-      "51X",
-      "Macpherson, Sir William (Alan)",
-      Some("Sir William Macpherson of Cluny (and Blairgowrie)"),
-      Some(1926),
-      None
-    ),
-    Person("515", "Newcastle, 1st Duke of, ", None, None, None),
-    Person("4VF", "Old Pretender, The", None, None, None),
-    Person("4H3", "Oliphant, Sir Mark Marcus Laurence Elwin", Some("Knight"), Some(1901), Some(2000)),
-    Person("46W", "Paine, Thomas", None, Some(1737), Some(1809)),
-    Person("3SH", "Reade, Hubert Granville Revell", None, Some(1859), Some(1938)),
-    Person("2TF", "Reading, ", None, None, None),
-    Person("53T", "Salisbury, Sir Edward James", Some("Professor"), Some(1886), Some(1978)),
-    Person("3QL", "Tate, Sir Henry", Some("1st Baronet"), Some(1819), Some(1899)),
-    Person("37K", "Uvarov, Sir Boris Petrovitch", Some("Knight"), Some(1889), Some(1970)),
-    Person("2T1", "Vane-Tempest-Stewart, Charles Stewart", Some("6th Marquess of Londonderry"), Some(1852), Some(1915)),
-    Person("4RW", "Victor Amadeus, ", Some("Duke of Savoy; King of Sardinia; King of Sicily"), Some(1666), Some(1732)),
-    Person("3GY", "Victoria, ", Some("Queen of Great Britain and Ireland"), Some(1819), Some(1901))
-  )
-
-  def getCreators: Seq[Creator] =
-    (getCorporateBodies.flatMap(Creator.from) ++ getPersons.flatMap(Creator.from)).sortBy(_.name)
+  def getCreators(): IO[Seq[Creator]] =
+    for {
+      persons         <- apiConnector.getPersons(GetPersons(timestamp = timeProvider.now()))
+      corporateBodies <- apiConnector.getCorporateBodies(GetCorporateBodies(timestamp = timeProvider.now()))
+    } yield persons.flatMap(Creator.from) ++ corporateBodies.flatMap(Creator.from)
 
   def getPlacesOfDeposit(): IO[Seq[PlaceOfDeposit]] =
     apiConnector.getPlacesOfDeposit(GetPlacesOfDeposit(timestamp = timeProvider.now()))
 
   def getLegalStatuses: IO[Seq[LegalStatus]] =
     apiConnector.getLegalStatuses(GetLegalStatuses(timeProvider.now()))
 
-  def isCreatorRecognised(creatorID: String): Boolean =
-    creatorID.trim.nonEmpty && getCreators.exists(_.id == creatorID)
+  def isCreatorRecognised(creators: Seq[Creator], creatorID: String): Boolean =
```

Comment:
I had moved a similar function for places of deposit to the only controller where it was needed:` EditSetRecordController`. It seems more at home there.

Only a suggestion.

Reply:
OK, moving this method also creates a stubbing exception.

Reply:
Let's look into it; will likely be easy to resolve.

Reply:
Confirmed.